File: Backend/src/main/java/com/app_template/App_Template/dto/CheckoutRequest.java
package com.app_template.App_Template.dto;

import com.app_template.App_Template.enums.PaymentMethod;
import com.app_template.App_Template.enums.ShippingMethod;
import lombok.Data;

import java.util.List;

@Data
public class CheckoutRequest {
    private String customerName;
    private String customerPhone;
    private String customerEmail;
    private String shippingAddress;
    private String billingAddress;
    private PaymentMethod paymentMethod;
    private ShippingMethod shippingMethod;
    private List<CartItemDto> cartItems;
}

File: Frontend/src/app/models/checkout-request.ts
export interface CheckoutRequest {
  customerName: string;
  customerPhone: string;
  customerEmail: string;
  shippingAddress: string;
  billingAddress: string;
  paymentMethod: 'CASH_ON_DELIVERY' | 'CARD_ONLINE';
  shippingMethod: 'FAN_COURIER' | 'SAME_DAY' | 'URGENT';
  cartItems: CartItemDto[];
}

export interface CartItemDto {
  id: number;
  productId: number;
  productName: string;
  productImage: string;
  productPrice: number;
  quantity: number;
  addedAt: string;
}
